orion/sites/admin/src/achieve/achieveEdit.js

```diff
@@ -3,31 +3,101 @@ require('./achieveEdit.styl');
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { Link } from 'react-router-dom';
+import API from '../api.js';
 
 export class AchieveEditPage extends React.Component {
     constructor(props) {
          super(props);
          this.state = {
-             inputYear: "",
-             inputMessage: "",
+             isEdit: false,
+             inputYear: 0,
+             inputMessage: ""
           };
+
+    this.handleChange = this.handleChange.bind(this);
+
+    this.onClickCancel = this.onClickCancel.bind(this);
+    this.onClickDelete = this.onClickDelete.bind(this);
+    this.onClickSave = this.onClickSave.bind(this);
+
+    this.onDeleted = this.onDeleted.bind(this);
+    this.onSaved = this.onSaved.bind(this);
+
     }
 
-    onClickSave() {
-        console.log("save button clicked");
+    componentDidMount() {
+      const achieveId = this.props.achieveId;
+      if (achieveId) {
+        API.get("api/achievements/v1/achievement/" + achieveId)
+          .then(res => {
+            debugger;
+            const achieve = res.data;
+            this.setState({
+              inputYear: achieve.year,
+              inputMessage: achieve.message,
+              isEdit: true
+            });
+          });
+      }
+    }
+
+    handleChange(event, value) {
```

Comment:
`value` is supposed to refer to `inputYear` or `inputMessage`.
Just having `value` doesn't really convey what this could be. Maybe give it a more specific name like `inputField`?